lib/active_interaction/base.rb

```diff
@@ -214,13 +215,35 @@ def inputs
     #   Example.run!(x: nil)  # => true
     #   Example.run!(x: rand) # => true
     #
+    # @example Nested checks
+    #   class Example < ActiveInteraction::Base
+    #     hash :x, default: {} do
+    #       integer :y, default: nil
+    #     end
+    #     def execute; given?(:x, :y) end
+    #   end
+    #   Example.run!()               # => false
+    #   Example.run!(x: nil)         # => false
+    #   Example.run!(x: {})          # => false
+    #   Example.run!(x: { y: rand }) # => true
```

Comment:
I think we should have an additional example where `y` is `nil` to really drive home what `given?` does.